### PR TITLE
Testcase spam (fix concurrency)

### PR DESCRIPTION
**Pull request by maxcai314**
This pull request adds test cases for Java's concurrency features such as park/unpark, interruption, and object monitors, and implements them with the scheduler.

This branch also adds a macro for `DEBUG_PEDANTIC_YIELD` which can be enabled by setting `DO_PEDANTIC_YIELDING` to 1, which will add a bunch of yield statements in various different async functions to test whether the VM handles all edge cases of continuation frames and futures.

bjvm should now be able to run all sorts of industrial concurrency utils!

### CODE REVIEW

**Comment by maxcai314** [comment]
This branch also cleans up a few of the the continuation/futures in the interpreter, eliminating a few unnecessary mallocs.

File: natives/share/java/lang/Thread.c [hunk 1]
```diff
@@ -71,12 +74,17 @@ DECLARE_ASYNC_NATIVE("java/lang", Thread, sleepNanos0, "(J)V", locals(rr_wakeup_
     ASYNC_RETURN_VOID();
   }
 
-  u64 end = get_unix_us() + nanos / 1000;
+  u64 start_us = get_unix_us(), sleep_us = (u64) nanos / 1000, end; // haha stupid three-in-one declaration
```
Comment (anematode) [inline natives/share/java/lang/Thread.c:77]: remove cmt?

**Review by anematode — CHANGES_REQUESTED** [review]
just the one thing :) then feel free to merge
